website/docs/d/google_kms_secret.html.markdown

```diff
@@ -0,0 +1,93 @@
+---
+layout: "google"
+page_title: "Google: google_kms_secret"
+sidebar_current: "docs-google-kms-secret"
+description: |-
+  Provides access to secret data encrypted with Google Cloud KMS
+---
+
+# google\_kms\_secret
+
+This data source allows you to use data encrypted with Google Cloud KMS
+within your resource definitions.
+
+For more information see
+[the official documentation](https://cloud.google.com/kms/docs/encrypt-decrypt).
+
+~> **NOTE**: Using this data provider will allow you to conceal secret data within your
+resource definitions, but it does not take care of protecting that data in the
+logging output, plan output, or state output.  Please take care to secure your secret
+data outside of resource definitions.
+
+## Example Usage
+
+First, create a KMS KeyRing and CryptoKey using the resource definitions:
+
+```hcl
+resource "google_kms_key_ring" "my_key_ring" {
+  project  = "my-project"
+  name     = "my-key-ring"
+  location = "us-central1"
+}
+
+resource "google_kms_crypto_key" "my_crypto_key" {
+  name     = "my-crypto-key"
+  key_ring = "${google_kms_key_ring.my_key_ring.id}"
+}
+```
+
+Next, use the [Cloud SDK](https://cloud.google.com/sdk/gcloud/reference/kms/encrypt) to encrypt some
+sensitive information:
+
+```bash
+$ echo -n my-secret-password | gcloud kms encrypt \
+> --project my-project \
+> --location us-central1 \
+> --keyring my-key-ring \
+> --key my-crypto-key \
+> --plaintext-file - \
+> --ciphertext-file - \
+> | base64
+CiQAqD+xX4SXOSziF4a8JYvq4spfAuWhhYSNul33H85HnVtNQW4SOgDu2UZ46dQCRFl5MF6ekabviN8xq+F+2035ZJ85B+xTYXqNf4mZs0RJitnWWuXlYQh6axnnJYu3kDU=
+```
+
+Finally, reference the encrypted ciphertext in your resource definitions:
+
+```hcl
+data "google_kms_secret" "sql_user_password" {
+  crypto_key = "${google_kms_crypto_key.my_crypto_key.id}"
+  ciphertext = "CiQAqD+xX4SXOSziF4a8JYvq4spfAuWhhYSNul33H85HnVtNQW4SOgDu2UZ46dQCRFl5MF6ekabviN8xq+F+2035ZJ85B+xTYXqNf4mZs0RJitnWWuXlYQh6axnnJYu3kDU="
```

Comment:
Nitpick: maybe this base64 text should be truncated? It's going to spill out of the `<code></code>` container on the doc site ... 

`ciphertext = "C1QAqD+x ... "` would probably do.

Reply:
I agree that you'd probably have to scroll right a bunch to see the whole thing, but I think that a complete example without truncated strings is worth that cost.

The [docs](https://www.terraform.io/docs/providers/aws/d/kms_secret.html) for `aws_kms_secret` also displays the entire encrypted string.

Reply:
I could, however, encrypt a shorter string.  That might make it easier to read.